pkg/asset/installconfig/aws/endpoint.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"fmt"
+
+	awsv2 "github.com/aws/aws-sdk-go-v2/aws"
+	ec2v2 "github.com/aws/aws-sdk-go-v2/service/ec2"
+	elbv2 "github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2"
+	iamv2 "github.com/aws/aws-sdk-go-v2/service/iam"
+	tagsv2 "github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi"
+	route53v2 "github.com/aws/aws-sdk-go-v2/service/route53"
+	s3v2 "github.com/aws/aws-sdk-go-v2/service/s3"
+	stsv2 "github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/sirupsen/logrus"
+)
+
+type endpointOptionInterface interface {
+	GetResolvedRegion() string
+	GetDisableHTTPS() bool
+	GetUseDualStackEndpoint() awsv2.DualStackEndpointState
+	GetUseFIPSEndpoint() awsv2.FIPSEndpointState
+}
+
+type endpointOptions struct {
+	ResolvedRegion       string
+	DisableHTTPS         bool
+	UseDualStackEndpoint awsv2.DualStackEndpointState
+	UseFIPSEndpoint      awsv2.FIPSEndpointState
+}
+
+func (e endpointOptions) GetResolvedRegion() string {
+	return e.ResolvedRegion
+}
+func (e endpointOptions) GetDisableHTTPS() bool { return e.DisableHTTPS }
+func (e endpointOptions) GetUseDualStackEndpoint() awsv2.DualStackEndpointState {
+	return e.UseDualStackEndpoint
+}
+func (e endpointOptions) GetUseFIPSEndpoint() awsv2.FIPSEndpointState {
+	return e.UseFIPSEndpoint
+}
+
+// getDefaultServiceEndpoint will get the default service endpoint for a service and region.
+func getDefaultServiceEndpoint(region, service string, opts endpointOptions) (*awsv2.Endpoint, error) { //nolint: staticcheck
```

Comment:
Do we need this? I'm not sure why we need to set the endpoints with default service endpoints. The SDK will use the default endpoints if they are unset?

It looks like we're only using this in order to determine the partition for the region, so it would make sense to make this more specific, like `getPartitionForRegion` and then needing to use the default ec2 resolver endpoint is just an implementation detail.